File: Core/src/main/java/jokatu/game/event/dialog/MainDialogRequestBuilder.java
package jokatu.game.event.dialog;

import jokatu.game.input.Input;
import jokatu.game.player.Player;
import jokatu.ui.Form;
import ophelia.builder.Builder;
import org.jetbrains.annotations.NotNull;

public interface MainDialogRequestBuilder<P extends Player, I extends Input> extends Builder<DialogRequest<P, I>, MainDialogRequestBuilder<P, I>> {

	@NotNull
	MainDialogRequestBuilder<P, I> withForm(@NotNull Form form);

	@NotNull
	MainDialogRequestBuilder<P, I> withoutCancelButton();
}

File: Core/src/main/kotlin/jokatu/game/event/dialog/MainDialogRequestBuilder.kt
package jokatu.game.event.dialog

import jokatu.game.input.Input
import jokatu.game.player.Player
import jokatu.ui.Form
import ophelia.builder.Builder

interface MainDialogRequestBuilder<P : Player, I : Input> : Builder<DialogRequest<P, I>, MainDialogRequestBuilder<P, I>> {

	fun withForm(form: Form): MainDialogRequestBuilder<P, I>

	fun withoutCancelButton(): MainDialogRequestBuilder<P, I>
}
